src/core/time_arrays.jl

```diff
@@ -478,26 +577,96 @@ function Base.setindex!(arr::TimestepArray{VariableTimestep{TIMES}, T, N, ti}, v
 	setindex!(arr.data, val, idxs1..., t, idxs2...)
 end
 
+# Indexing with arrays of TimestepIndexes or TimestepValues
+
+function Base.getindex(arr::TimestepArray{FixedTimestep{FIRST, STEP}, T, N, ti}, idxs::Union{Array{TimestepIndex,1}, AnyIndex}...) where {T, N, ti, FIRST, STEP}
+	idxs1, ts_array, idxs2 = split_indices(idxs, ti)
+	ts_idxs = _get_ts_indices(ts_array)
+	return arr.data[idxs1..., ts_idxs, idxs2...]
+end
+
+function Base.getindex(arr::TimestepArray{VariableTimestep{TIMES}, T, N, ti}, idxs::Union{Array{TimestepIndex,1}, AnyIndex}...) where {T, N, ti, TIMES}
+	idxs1, ts_array, idxs2 = split_indices(idxs, ti)
+	ts_idxs = _get_ts_indices(ts_array)
+	return arr.data[idxs1..., ts_idxs, idxs2...]
+end
```

Comment:
Can these first two functions be just one? Seems like it's the same whether the TimestepArray is parameterized by a FixedTimestep or a VariableTimestep

```
function Base.getindex(arr::TimestepArray{TS, T, N, ti}, idxs::Union{Array{TimestepIndex,1}, AnyIndex}...) where {TS, T, N, ti}
	idxs1, ts_array, idxs2 = split_indices(idxs, ti)
	ts_idxs = _get_ts_indices(ts_array)
	return arr.data[idxs1..., ts_idxs, idxs2...]
end
```
I might be wrong though about if it will let you do that...

Reply:
I think you're probably right!  I was trying to simplify to ` TS where {TS <: AbstractTimestep}` in a few places but sometimes it causes errors so I wanted to get it working first.  I'll try it now that it's all working.